mappings/muse.py

```diff
@@ -1,100 +1,170 @@
+            table_of_contents=toc[0] if toc else  None,
+            abstract=self._get_abstracts(muse_record),
+            subjects=self._get_subjects(muse_record),
+            contributors=self._get_contributors(muse_record),
+            is_part_of=self._get_formatted_field(muse_record, '490', ['a','v'], '{}|{}|volume'),
+            has_part=self._get_formatted_field(muse_record, '856', ['u'], '1|{}|muse|text/html|{{"reader": false, "download": false, "catalog": false, "embed": true}}'),
+            rights=rights,
+            date_created=datetime.now(timezone.utc).replace(tzinfo=None),
+            date_modified=datetime.now(timezone.utc).replace(tzinfo=None)
+        )
 
-    def applyFormatting(self):
-        self.record.source = 'muse'
-        self.record.source_id = list(self.record.identifiers[0].split('|'))[0]
+        return new_record
+
+    def _get_formatted_field(self, record, field, subfields, string_format):
+        field_data = record.get_fields(field)
+        formatted_data = []
+        subfield_data = []
+
+        for data in field_data:
+            if data.is_control_field():
+                subfield_data.append([getattr(data, 'data')])
+            else:
+                subfield_data = [data.get_subfields(subfield) if data.get_subfields(subfield) else [''] for subfield in subfields]
+            
+            if subfield_data == [['']]:
+                continue
 
-        # Take first title as they are in order of preference
-        self.record.title = self.record.title[0]
+            for values in zip(*subfield_data):
+                formatted_string = string_format.format(*values)
+                formatted_data.append(utils.clean_formatted_string(formatted_string))
 
-        self.record.identifiers = [self.cleanup_identifier(id) for id in self.record.identifiers]
+        return formatted_data
+    
+    def _get_identifiers(self, record):
+        all_identifiers = []
         
-        self.record.languages = [
+        all_identifiers.extend(self._get_formatted_field(record, '001', ['0'], '{}|muse'))
+        all_identifiers.extend(self._get_formatted_field(record, '010', ['z'], '{}|lccn'))
+        all_identifiers.extend(self._get_formatted_field(record, '020', ['a', 'z'], '{}{}|isbn'))
+        all_identifiers.extend(self._get_formatted_field(record, '022', ['a'], '{}|issn'))
+        all_identifiers.extend(self._get_formatted_field(record, '035', ['a'], '{}|oclc'))
+        all_identifiers.extend(self._get_formatted_field(record, '040', ['a'], '{}|muse'))
+
+        return [self._cleanup_identifier(id) for id in all_identifiers]
+    
+    def _get_authors(self, record):
+        all_authors = []
+
+        all_authors.extend(self._get_formatted_field(record, '100', ['a', 'b', 'c', 'd'], '{} {} {} {}|||true'))
+        all_authors.extend(self._get_formatted_field(record, '110', ['a', 'b', 'c', 'n', 'd'], '{} {} {} {} {}|||true'))
+
+        return all_authors
```

Comment:
up to you but you could also do 

```
return [
  self._get_formatted_field(record, '100', ['a', 'b', 'c', 'd'], '{} {} {} {}|||true'),
  self._get_formatted_field(record, '110', ['a', 'b', 'c', 'n', 'd'], '{} {} {} {} {}|||true')
]
```

more of a style choice but it removes the all_authors array variable and shortens up the line length.

Reply:
Updated this to have a separate helper method _get_formatted_fields that calls _get_formatted_field for a list of passed in field and string formats with the subfields.